rasa_core/processor.py

```diff
@@ -337,7 +337,17 @@ def _schedule_reminders(self, events: List[Event],
                                       args=[e, dispatcher],
                                       id=e.name,
                                       replace_existing=True,
-                                      name=str(e.action_name))
+                                      name=str(e.action_name) + "__sender_id:" + tracker.sender_id)
+
+    def _cancel_reminders(self, events: List[Event], tracker) -> None:
+        # All Reminders with the same name will be cancelled
+        if events is not None:
+            for e in events:
+                if isinstance(e, ReminderCancelled):
```

Comment:
let's also create `id_to_check = e.name` if it is not `None` cancel by `id` otherwise look for `action_name` and cancel by it

Reply:
Can you please explain that in more detail?

Reply:
reminder can be identified by its `name` (the same as `id`) or by the name of the action it triggers, I suggest if user specified `name` to cancel let's cancel by the name otherwise find it by `action_name` for the same `sender_id`. Does it make sense?

Reply:
The name is always set to a unique identifier: _action_name + "__sender_id:" + sender_id_
Therefore I guess it makes only sense to use the name, otherwise jobs from other users with the same name would be canceled as well.

Reply:
Yes, I guess it doesn't make sense to cancel reminder, created for all users from this event